src/main/java/seedu/address/ui/BrowserPanel.java

```diff
@@ -42,70 +43,117 @@
     private WebEngine webEngine = browser.getEngine();
 
     private String currentLocation;
-    private boolean isCurrentlyNonBrowserView;
+    private boolean isCurrentlyReaderView;
     private ViewMode viewMode;
+    private Consumer<CommandResult> onSuccess;
```

Comment:
We'll merge this for now but note this is why we usually don't want to do any processing on the UI side :p

Reply:
The most recent commit doesn't have the CommandResult consumer any more.